app/models/match.rb

```diff
@@ -2,26 +2,32 @@
 #
 # Table name: matches
 #
-#  id              :uuid             not null, primary key
-#  starts_at       :datetime
-#  casters         :jsonb
-#  data            :jsonb
-#  away_team_id    :uuid             not null
-#  home_team_id    :uuid             not null
-#  losing_team_id  :uuid
-#  winning_team_id :uuid
-#  season_id       :uuid             not null
-#  created_at      :datetime         not null
-#  updated_at      :datetime         not null
+#  id           :uuid             not null, primary key
+#  starts_at    :datetime
+#  casters      :jsonb
+#  data         :jsonb
+#  away_team_id :uuid             not null
+#  home_team_id :uuid             not null
+#  loser_id     :uuid
+#  winner_id    :uuid
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
 #
 
 class Match < ApplicationRecord
-  belongs_to :season
+  has_one :division_match
+  has_one :division, through: :division_match
 
-  belongs_to :away_team
-  belongs_to :home_team
-  belongs_to :loser,  foreign_key: :losing_team_id
-  belongs_to :winner, foreign_key: :winning_team_id
+  has_one :season, through: :division
 
-  has_many   :games
+  belongs_to :away_team, class_name: 'Team'
+  belongs_to :home_team, class_name: 'Team'
+
+  belongs_to :loser,  class_name: 'Team', optional: true
```

Comment:
Simplified to `loser` and `winner` (let's call these what they are), and included `optional` here as the default behavior of foreign keys changed in Rails 5:

https://github.com/rails/rails/issues/18233